### PR TITLE
Modernize codebase to pass rust-2018-idioms lints

### PR DESCRIPTION
**Pull request by mlafeldt**
By adding the lint rule and running `clippy fix`.

As far as I can see, all of the changes fix `error: hidden lifetime parameters in types are deprecated` aka `-D elided-lifetimes-in-paths`, which is implied by `-D rust-2018-idioms`.

### CODE REVIEW

File: tools/lint.js [hunk 1]
```diff
@@ -120,6 +120,8 @@ async function clippy() {
       "clippy::missing_safety_doc",
       "-D",
       "clippy::undocumented_unsafe_blocks",
+      "-D",
+      "rust-2018-idioms",
```
Comment (mlafeldt) [inline tools/lint.js:124]: Needs to be moved to cargo config should #15453 be accepted first.

Comment (mlafeldt) [reply tools/lint.js]: Rebased

File: cli/args/config_file.rs [hunk 2]
```diff
@@ -66,7 +66,7 @@ pub struct IgnoredCompilerOptions {
 }
 
 impl fmt::Display for IgnoredCompilerOptions {
-  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```
Comment (dsherret) [inline cli/args/config_file.rs:69]: Kind of seems like a step back. I like hidden lifetimes.

Comment (mlafeldt) [reply cli/args/config_file.rs:69]: OK, if that's a project-wide decision, what about making it explicit? e.g.

```diff
diff --git .cargo/config.toml .cargo/config.toml
index cac99a377..f52833b14 100644
--- .cargo/config.toml
+++ .cargo/config.toml
@@ -23,4 +23,8 @@ rustflags = [
   "clippy::missing_safety_doc",
   "-D",
   "clippy::undocumented_unsafe_blocks",
+  "-D",
+  "rust-2018-idioms",
+  "-A",
+  "elided-lifetimes-in-paths",
 ]
```

Or maybe even just `-A elided-lifetimes-in-paths` if `rust-2018-idioms` isn't deemed useful.

**Pull request closed without merging**